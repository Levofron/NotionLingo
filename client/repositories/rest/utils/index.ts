export * from './format-word-suggestions/format-word-suggestions.function';
export * from './format-random-notion-words-transformator/format-random-notion-words-transformator.function';
export * from './add-image-url-for-each-notion-word-transformator/add-image-url-for-each-notion-word-transformator.function';
