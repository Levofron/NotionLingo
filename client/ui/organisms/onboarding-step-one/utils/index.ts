export * from './map-column-names-to-components/map-column-names-to-components.function';
