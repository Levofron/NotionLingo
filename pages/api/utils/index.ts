export { decrypt } from './decrypt/decrypt.function';
export { encrypt } from './encrypt/encrypt.function';
export { withMiddleware } from './with-middleware/with-middleware.function';
export { createNotionClient } from './create-notion-client/create-notion-client.function';
export { validateIfUserIsLoggedIn } from './validate-if-user-is-logged-in/validate-if-user-is-logged-in.function';
export { validateRouteSecretMiddleware } from './validate-route-secret-middleware/validate-route-secret-middleware.function';
export { validateRequestMethodMiddleware } from './validate-request-method-middleware/validate-request-method-middleware.function';
export { assignRequestTokenToSupabaseSessionMiddleware } from './assign-request-token-to-supabase-session-middleware/assign-request-token-to-supabase-session-middleware.function';
