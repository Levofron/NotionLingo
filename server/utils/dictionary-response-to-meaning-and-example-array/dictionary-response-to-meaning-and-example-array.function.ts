import { IDictionaryResponse } from './dictionary-response-to-meaning-and-example-array.types';

export const dictionaryResponseToMeaningAndExampleArray = ({
  additionalExamples,
  meaningAndExamples,
  word,
}: IDictionaryResponse) => {
  const suggestions = meaningAndExamples.map(({ examples, meaning }) => ({
    meaning,
    example: examples?.length ? examples[0] : additionalExamples[0] || '',
  }));

  return {
    word,
    suggestions,
  };
};
