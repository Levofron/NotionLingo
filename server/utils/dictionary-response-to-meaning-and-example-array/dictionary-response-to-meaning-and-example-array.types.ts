export interface IMeaningWithExamples {
  examples: string[];
  meaning: string;
}

export interface IDictionaryResponse {
  additionalExamples: string[];
  meaningAndExamples: IMeaningWithExamples[];
  word: string;
}
