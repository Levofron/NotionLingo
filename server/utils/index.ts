export * from './decrypt/decrypt.function';
export * from './encrypt/encrypt.function';
export * from './text-to-ipa/text-to-ipa.function';
export * from './shuffle-array/shuffle-array.function';
export * from './with-middleware/with-middleware.function';
export * from './get-profile-by-id/get-profile-by-id.function';
export * from './get-random-number/get-random-number.function';
export * from './create-notion-client/create-notion-client.function';
export * from './get-user-from-request/get-user-from-request.function';
export * from './get-notion-table-columns/get-notion-table-columns.function';
export * from './get-available-notion-databases/get-available-notion-databases.function';
export * from './is-valid-notion-database-schema/is-valid-notion-database-schema.function';
export * from './validate-route-secret-middleware/validate-route-secret-middleware.function';
export * from './validate-request-method-middleware/validate-request-method-middleware.function';
export * from './get-text-from-page-property-instance/get-text-from-page-property-instance.function';
export * from './get-profile-data-with-notion-data-check/get-profile-data-with-notion-data-check.function';
export * from './validate-if-user-is-logged-in-middleware/validate-if-user-is-logged-in-middleware.function';
export * from './validate-if-parameters-exists-middleware/validate-if-parameters-exists-middleware.function';
export * from './assign-request-token-to-supabase-session-middleware/assign-request-token-to-supabase-session-middleware.function';
