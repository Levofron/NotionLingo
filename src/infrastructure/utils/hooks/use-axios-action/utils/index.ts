export * from './is-axios-error/is-axios-error.function';
