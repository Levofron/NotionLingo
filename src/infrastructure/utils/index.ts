export * from './jest';
export * from './hooks';
export * from './guards';
export * from './functions';
