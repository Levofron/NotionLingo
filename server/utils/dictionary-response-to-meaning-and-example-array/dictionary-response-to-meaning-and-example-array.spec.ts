import { functionImportTest } from '@infrastructure/utils';

import { dictionaryResponseToMeaningAndExampleArray } from './dictionary-response-to-meaning-and-example-array.function';

describe('dictionaryResponseToMeaningAndExampleArray function', () => {
  functionImportTest(dictionaryResponseToMeaningAndExampleArray);

  it('should return same object with cleaned up strings', () => {
    const data = {
      additionalExamples: ['example 3', 'example 24'],
      meaningAndExamples: [
        {
          meaning: 'meaning 1',
          examples: ['example 1', 'example 2'],
        },
        {
          meaning: 'meaning 2',
          examples: ['example 1', 'example 2'],
        },
        {
          examples: [],
          meaning: 'meaning 3',
        },
      ],
      word: 'word',
    };

    const response = dictionaryResponseToMeaningAndExampleArray(data);

    expect(response).toMatchSnapshot();
  });
});
