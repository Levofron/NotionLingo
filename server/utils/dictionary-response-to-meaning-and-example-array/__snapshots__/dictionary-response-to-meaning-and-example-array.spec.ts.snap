// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dictionaryResponseToMeaningAndExampleArray function should return same object with cleaned up strings 1`] = `
{
  "meaningsAndExamples": [
    {
      "example": "example 1",
      "meaning": "meaning 1",
    },
    {
      "example": "example 1",
      "meaning": "meaning 2",
    },
    {
      "example": "example 3",
      "meaning": "meaning 3",
    },
  ],
  "word": "word",
}
`;
